### PR TITLE
Mind footer/header bounds when navigating vertically

### LINKED ISSUE #1025: Focused cell can get hidden under the footer when navigating
screenshot 2017-10-04 16 56 49

### PR DESCRIPTION
**Pull request by tomivirkki**
Fixes #1025

---
This change is [Reviewable](https://reviewable.io/reviews/vaadin/vaadin-grid/1026)

### CODE REVIEW

**Comment by platosha** [comment]
Reviewed 2 of 2 files at r1.
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---

*[vaadin-grid-keyboard-navigation-mixin.html, line 296 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-grid/1026#-Kvg3h11Vk0lcR6NqVxd:-Kvg3h12Neoa4USs4URg:b-22neyl) ([raw file](https://github.com/vaadin/vaadin-grid/blob/ea1c6a4d337756a7e8de94230fe8c18b0b6b2738/vaadin-grid-keyboard-navigation-mixin.html#L296)):*
> ```HTML
>       }
> 
>       const dstRect = dstCell.getBoundingClientRect();
> ```

Well, I would expect `_ensureScrolledToIndex` to handle that.

In other words, once `_firstVisibleIndex` and `_lastVisibleIndex` account for that, `_scrollToIndex(dstRowIndex)` should be enough, we don’t have to apply further adjustments to scroll position after that.

So should can we fix `_firstVisibleIndex` and `_lastVisibleIndex` instead?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-grid/1026)*

**Comment by tomivirkki** [comment]
*[vaadin-grid-keyboard-navigation-mixin.html, line 296 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-grid/1026#-Kvg3h11Vk0lcR6NqVxd:-Kvg57KY0pRCljsBMylW:b6qilkb) ([raw file](https://github.com/vaadin/vaadin-grid/blob/ea1c6a4d337756a7e8de94230fe8c18b0b6b2738/vaadin-grid-keyboard-navigation-mixin.html#L296)):*
<details><summary><i>Previously, platosha (Anton Platonov) wrote…</i></summary><blockquote>

Well, I would expect `_ensureScrolledToIndex` to handle that.

In other words, once `_firstVisibleIndex` and `_lastVisibleIndex` account for that, `_scrollToIndex(dstRowIndex)` should be enough, we don’t have to apply further adjustments to scroll position after that.

So should can we fix `_firstVisibleIndex` and `_lastVisibleIndex` instead?
</blockquote></details>

last/first visible index are iron-list properties and I think both of them already work according to spec.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-grid/1026)*

**Comment by platosha** [comment]
:lgtm:

---

Review status: :shipit: all files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-grid/1026#-:-KvgSx6eYIOcW3YU0DzF:bnfp4nl)*
